qlkube/src/decorateSubscription.js

```diff
@@ -92,29 +106,79 @@ function decorateSubscription(baseSchema, targetType, enumType) {
   return newSchema;
 }
 
-function setupSubscriptions(subscriptions, schema) {
+async function checkPermission(
+  token,
+  group = '',
+  resource,
+  namespace,
+  name = ''
+) {
+  if (!token) throw 'Parameter token cannot be empty!';
+  if (!resource) throw 'Parameter resource cannot be empty!';
+  if (!namespace) throw 'Parameter namespace cannot be empty!';
+
+  const keyCache = `${token}_${group}_${resource}_${namespace}_${name}`;
+  const lastSub = cacheSubscriptions[keyCache];
+  const canUserWatchResourceCached =
+    lastSub &&
+    !(
+      Date.now() - lastSub > 10 * 60 * 1000 &&
```

Comment:
Define a constant `TEN_SECONDS` at the start of the file to make the code more readable. (do it for other places where you compute a certain amount of time)